### PR TITLE
[Swift5] replace special characters in Swift enum var name 

### PR DESCRIPTION
**Pull request by mbruegmann**
A fix for this issue: https://github.com/OpenAPITools/openapi-generator/issues/11107

@jgavris (2017/07) @ehyche (2017/08) @Edubits (2017/09) @jaz-ah (2017/09) @4brunu (2019/11)

### PR checklist
 
- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [ ] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [ ] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by 4brunu** [comment]
Hi @mbruegmann, first of all thanks for creating this PR 👍 
Could you please run the following commands and commit the output please?
```
./mvnw clean package 
./bin/generate-samples.sh
./bin/utils/export_docs_generators.sh
```
Thanks

File: samples/client/petstore/swift5/resultLibrary/PetstoreClient/Classes/OpenAPIs/Models/EnumArrays.swift [hunk 1]
```diff
@@ -13,7 +13,7 @@ import AnyCodable
 public struct EnumArrays: Codable, Hashable {
 
     public enum JustSymbol: String, Codable, CaseIterable {
-        case greaterThanOrEqualTo = ">="
+        case greaterThanEqual = ">="
```
Comment (4brunu) [inline samples/client/petstore/swift5/resultLibrary/PetstoreClient/Classes/OpenAPIs/Models/EnumArrays.swift:16]: @mbruegmann I think those changes are not expected, or are they?
Could you please check what happened?

Comment (mbruegmann) [reply samples/client/petstore/swift5/resultLibrary/PetstoreClient/Classes/OpenAPIs/Models/EnumArrays.swift:16]: I hav no idea what this is for. Those were the only changes I got after executing the given 3 commands.

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/swift5/Swift5ClientCodegenTest.java [hunk 2]
```diff
@@ -95,6 +95,15 @@ public void testStartingWithNumber() throws Exception {
         Assert.assertEquals(swiftCodegen.toEnumVarName("123EntryName123", null), "_123entryName123");
     }
 
+    @Test(enabled = true)
+    public void testSpecialCharacters() throws Exception {
+        Assert.assertEquals(swiftCodegen.toEnumVarName("1:1", null), "_1Colon1");
```
Comment (4brunu) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/swift5/Swift5ClientCodegenTest.java:100]: @mbruegmann the previous changes are related to the changes you make in the Swift5ClientCodegen.java.
An easy way to test it is by adding the following test case here that should pass.
```
Assert.assertEquals(swiftCodegen.toEnumVarName(">=", null), "greaterThanOrEqualTo");
```

Comment (mbruegmann) [reply modules/openapi-generator/src/test/java/org/openapitools/codegen/swift5/Swift5ClientCodegenTest.java:100]: It seems to be an rare edge case. I am doing the replacement character by character now to also replace special characters inside the name. Before only in case of a match of the whole string a replacement was done.
This was the previous code and it would not work for "3=>2" for example. It would only match "=>".

`if (getSymbolName(name) != null) {
            return camelize(WordUtils.capitalizeFully(getSymbolName(name).toUpperCase(Locale.ROOT)), true);
}`
The problem comes from `specialCharReplacements` in `DefaultCodegen`. It's a map with replacements that could be reversed, but the order also is not garanteed.

**Comment by 4brunu** [comment]
Could you please confirm that the following command doesn't change the sample projects?
```
./mvnw clean package 
./bin/generate-samples.sh
./bin/utils/export_docs_generators.sh
```
Thanks

**Comment by mbruegmann** [comment]
After running the 3 commands there were no changes.

**Review by 4brunu — APPROVED** [review]
Looks good to me 👍 
Thanks for creating this PR.
